core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/OAbstractPaginatedStorage.java

```diff
@@ -4001,12 +4020,19 @@ private ORawBuffer readRecordIfNotLatest(final OCluster cluster, final ORecordId
 
     stateLock.acquireReadLock();
     try {
+      if (pessimisticLock) {
+        acquireReadLock(rid);
+      }
+
       ORawBuffer buff;
       checkOpenness();
 
       buff = doReadRecordIfNotLatest(cluster, rid, recordVersion);
       return buff;
     } finally {
+      if (pessimisticLock) {
```

Comment:
You should check if the lock was acquired. If it wasn't it  throws a `OLockException` exception